src/vma/util/lock_wrapper.h

```diff
@@ -269,13 +269,13 @@ class lock_mutex : public lock_base
 	inline int lock() {
 		tscval_t t = start_lock_wait();
 		int ret = pthread_mutex_lock(&m_lock);
-		lock_base::lock();
+		if (0 == ret) ret = lock_base::lock();
```

Comment:
This might cause some performance hit.

The lock_base::lock() should be optimized out by default, since without LOCK_STATS it will statically 'return 0' and compiler remove that code.
Now there is a branch point which depends on 'ret', so i expect some penalty for each call.